standalone/pruntime/enclave/src/lib.rs

```diff
@@ -1361,15 +1490,46 @@ fn dispatch_block(input: DispatchBlockReq) -> Result<Value, Value> {
             .expect("ECDH not initizlied"),
     );
     let mut last_block = 0;
-    for block in blocks.iter() {
-        if block.events.is_none() {
-            return Err(error_msg("Event was required"));
+    for block in blocks.into_iter() {
+        let expected_root = local_state
+            .block_hashes
+            .get(0)
+            .ok_or(error_msg("No enough headers to validate the blocks"))?
+            .1;
+
+        let changes = &block.storage_changes;
+        local_state.runtime_state.apply_changes(
+            &changes.main_storage_changes,
+            &changes.child_storage_changes,
+        );
+
+        // TODO.kevin: rollback the state on failure
```

Comment:
Oh, I got your concern here. In substrate, a new state root is calculated on an overlay db. Then if the root matches the block header, it gets applied to the state db. If it doesn't match, the overlay db can be dropped.

If we need hours to fix this, let's just leave a TODO here and pay attention later. If it's easy enough, I suggest to make changes only after validating the new state root.